src/dreyfus_index.erl

```diff
@@ -164,23 +164,11 @@ handle_info({'EXIT', FromPid, {updated, NewSeq}},
             #state{
               index=Index0,
               index_pid=IndexPid,
-              updater_pid=UpPid,
-              waiting_list=WaitList
+              updater_pid=UpPid
              }=State) when UpPid == FromPid ->
     Index = Index0#index{current_seq=NewSeq},
-    case reply_with_index(IndexPid, Index, WaitList) of
```

Comment:
these lines are extracted to a new function to make it easier to understand.
I followed the comments in hastings from Paul Davis.